src/lwc/rd2EntryFormScheduleSection/rd2EntryFormScheduleSection.js

```diff
@@ -144,7 +287,133 @@ export default class rd2EntryFormScheduleSection extends LightningElement {
+            this.isAdvancedMode = true;
+            this.showDayOfMonth = (this.customPeriodAdvancedMode === PERIOD_MONTHLY);
+            this.scheduleRowColumnSize = (this.showDayOfMonth ? 3 : 4);
+
+            if (advancedPeriod === PERIOD_MONTHLY) {
+                this.showDayOfMonth = true;
+                this.scheduleRowColumnSize = 3;
+            } else {
+                this.showDayOfMonth = false;
+                this.scheduleRowColumnSize = 4;
+            }
+        }
+    }
+
+    /**
+     * @description Custom Period picklist options - Advanced and Monthly (using the correct labels)
+     */
+    get customPeriodOptions() {
+        let monthlyLabel = PERIOD_MONTHLY;
+
+        // Get the translated labels for Monthly if there is one
+        this.installmentPeriodPicklistValues
+            .forEach(pl => {
+                if (pl.value === PERIOD_MONTHLY) {
+                    monthlyLabel = pl.label;
+                }
+            });
+
+        return [
+            { label: monthlyLabel, value: PERIOD_MONTHLY },
+            { label: this.customLabels.picklistLabelAdvanced, value: RECURRING_PERIOD_ADVANCED },
+        ];
+    }
+
+    /**
+     * @description Build the picklist values to use for the Period picklist in the Advanced view. These replace
+     * the standard Monthly, Weekly, Yearly labels with Months, Weeks, Years - but only for labels visible in
+     * this Picklist on this UI.
+     */
+    get advancedPeriodOptions() {
+        let advancedPeriodPicklistValues = [];
+        this.installmentPeriodPicklistValues
+            .forEach(pl => {
+                switch (pl.value) {
+                    case PERIOD_DAILY:
+                        advancedPeriodPicklistValues.push(
+                            {label: this.customLabels.periodPluralDays, value: pl.value}
+                        );
+                        break;
+                    case PERIOD_WEEKLY:
+                        advancedPeriodPicklistValues.push(
+                            {label: this.customLabels.periodPluralWeeks, value: pl.value}
+                        );
+                        break;
+                    case PERIOD_MONTHLY:
+                        advancedPeriodPicklistValues.push(
+                            {label: this.customLabels.periodPluralMonths, value: pl.value}
+                        );
+                        break;
+                    case PERIOD_YEARLY:
+                        advancedPeriodPicklistValues.push(
+                            {label: this.customLabels.periodPluralYears, value: pl.value}
+                        );
+                        break;
+                    case PERIOD_FIRST_AND_FIFTEENTH:
+                        advancedPeriodPicklistValues.push(
+                            {label: pl.label, value: pl.value}
+                        );
+                        break;
+                }
+
+            });
+        return advancedPeriodPicklistValues;
     }
 
     /**
```

Comment:
isValid() should also report the validity on lightning-combobox to make sure Installment Period picklist is valid on save.
`this.template.querySelectorAll('lightning-combobox').forEach(.....)`